contributed/Source+Catalog+Emulation+with+CosmoDC2Realizer.ipynb

```diff
@@ -0,0 +1,535 @@
+    "              'size_bulge_true', 'size_minor_bulge_true', 'sersic_bulge', 'ellipticity_1_bulge_true',\n",
+    "              'ellipticity_2_bulge_true', 'ellipticity_bulge_true',\n",
+    "              'size_disk_true', 'size_minor_disk_true', 'sersic_disk', 'ellipticity_1_disk_true',\n",
+    "              'ellipticity_2_disk_true', 'ellipticity_disk_true',\n",
+    "              'bulge_to_total_ratio_i',\n",
+    "              'mag_true_u_lsst',\n",
+    "              'mag_true_g_lsst',\n",
+    "              'mag_true_r_lsst',\n",
+    "              'mag_true_i_lsst',\n",
+    "              'mag_true_z_lsst',\n",
+    "              'mag_true_Y_lsst',\n",
+    "              'halo_mass',]\n",
+    "\n",
+    "cuts = [# A loose magnitude cut\n",
+    "        GCRQuery('mag_true_g_lsst < 27'), \n",
+    "        # Query halo masses likely to host an AGN\n",
+    "        GCRQuery('halo_mass > 1.e13'),\n",
+    "        # Query sources belonging to Field 1188\n",
+    "        GCRQuery('((ra_true - %f)**2.0 + (dec_true - %f)**2.0)**0.5 < %f' %(field_ra, field_dec, field_radius)),\n",
+    "        #GCRQuery('abs(dec_true - %f) < %f' %(field_dec, field_radius)),\n",
+    "        ]\n",
+    "# Add filters as necessary!\n",
+    "galaxies = catalog.get_quantities(quantities, filters=cuts)\n",
+    "galaxies = pd.DataFrame(galaxies)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Getting line-of-sight neighbors\n",
+    "For each galaxy, any other galaxy or point source (unlensed AGN or star) that lie within its blending scale will be its line-of-sight neighbor. Galaxy neighbors will simply be taken from the extragalactic catalog, which we've already fetched. Point-source neighbors will be taken from the truth catalog as below.\n",
+    "\n",
+    "### Getting the neighbors (unlensed AGNs and stars) from the truth catalog"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# truth_catalog = GCRCatalogs.load_catalog('dc2_truth_run1.1_static')\n",
+    "truth_catalog = GCRCatalogs.load_catalog('dc2_truth_run1.1', {'md5': None})\n",
+    "\n",
+    "truth_catalog_columns = ['object_id', 'ra', 'dec', 'star', 'agn', 'sprinkled', 'healpix_2048',\n",
+    "                        'u', 'g', 'r', 'i', 'z', 'y',]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 44,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 31.6 s, sys: 4.25 s, total: 35.8 s\n",
+      "Wall time: 35.9 s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "# ~40 in Jupyter-dev\n",
+    "cuts = ['(star==1) | (agn==1)',\n",
+    "        'sprinkled==0',\n",
+    "        '((ra - %f)**2.0 + (dec - %f)**2.0)**0.5 < %f' %(field_ra, field_dec, field_radius),\n",
+    "       ]\n",
```

Comment:
Could the users specify the ra, dec range that they want to use for this? Or have it defined as a variable so that it can be changed/discussed earlier?

Reply:
I see this wasn't clear! I've mentioned in the markdown that the list of pointings can be expanded.